### PR TITLE
Add support for Molmo-D-7B Model

### PR DESCRIPTION
**Pull request by BabyChouSr**
Support for https://huggingface.co/allenai/Molmo-7B-D-0924

It's been awhile since I've supported a model, so I figured it would be good practice. It seems to be a CLIP-Vit encoder with some embedding pooling and then a Qwen LLM backbone. I'm trying to figure out the vision backbone first and then I think the frozen parts will come together by importing from Qwen2.py file.

### CODE REVIEW

**Comment by jlia0** [comment]
Any updates on this?

**Comment by BabyChouSr** [comment]
@jlia0 I'm working on this slowly since I'm juggling a couple things. It will take me probably a week.

File: test/srt/models/test_molmo.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                torch.max(abs(hf_outputs[0] - srt_outputs[0])) < 1e-3
+            ), "image_features are not all close"
+            assert (
+                torch.max(abs(hf_outputs[1] - srt_outputs[1])) < 1e-3
+            ), "cls_embed are not all close"
+
+    def test_molmo_d(self):
+        for text, images in CONVS:
+            for model_path, torch_dtype, tolerance in MODELS:
+                self.assert_srt_vision_backbone_and_hf_vision_backbone_close(
```
Comment (merrymercy) [inline test/srt/models/test_molmo.py:123]: Do you have an end-to-end test for the whole model?
Can you add a new test here https://github.com/sgl-project/sglang/blob/main/test/srt/test_vision_openai_server.py?

**Review by merrymercy — CHANGES_REQUESTED** [review]

**Comment by merrymercy** [comment]
Closed due to inactivity. Feel free to reopen if you have new progress.

**Pull request closed without merging**